src/mediawiki-adapter.ts

```diff
@@ -0,0 +1,44 @@
+import { mwn } from "mwn";
+export const sizeLimit:number = 256*1024
+
+export interface AudioInfo
+{
+    size: number
+    url: string
+    mediatype: string
+}
+
+interface queryResult
+{
+    query: {
+        pages: {
+            missing: boolean
+            imageinfo: AudioInfo[]
+        }[]
+    }
+}
+export async function searchAudioFile(domainApiUrl:string, filename: string) {
+    if (!domainApiUrl)
+        return null;
+    const bot = new mwn({ // constructing this way allows sending requests without login
+        apiUrl: domainApiUrl,
+        silent: false,      // suppress messages (except error messages)
+        retryPause: 5000,   // pause for 5000 milliseconds (5 seconds) on maxlag error.
+        maxRetries: 3       // attempt to retry a failing requests upto 3 times
```

Comment:
Does this mean there might be  up to 15 seconds delay for playback? I feel anything more than few seconds (3-to-5 maybe??) delay is not worth playing any sound at all.

Reply:
I would allow at least one retry with 1 second delay. I feel like it is not that uncommon to get a negative response from the api.